### PR TITLE
172 Improvement spotifyservice with arealoop

### LINKED ISSUE #172: [IMPROVEMENT]: SpotifyService with AreaLoop.
### Description

The Spotify service is not suitable for AreaLoop. 
Actions do not return a bool.
It does not contain a function linked to AreaLoop to execute actions & reactions.

### To do

Creating a function tree to execute actions in AreaLoop.
Create a function tree to execute reactions in AreaLoop.
Probably register the tokens in the DB (see with Nathan).
Link SpotifyService with AreaLoop.
Maybe remove au res & req not necessary.

### Priority

High

### Project parts

- [X] Server
- [ ] Application
- [ ] Web
- [ ] Automation

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 31Nathan**
# Updated feature
Spotify actions and reactions are now usable for Area Loop

### CODE REVIEW

File: Server/Services/spotifyServices.js [hunk 1]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
  * @requires querystring
  */
 const querystring  = require('querystring');
+const { resolve } = require('path');
+
+/**
+ * Checks if the logged user follows some artists
+ * @function isfollowing
+ * @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:69]: missing async field

File: Server/Services/spotifyServices.js [hunk 2]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
...
+    })
+    .catch(error =>{
+        console.error(error)
+    })
+}
+
+/**
+ * Checks if the logged user is currently listening to some music
+ * @function isListening
+ * @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:100]: missing async field

File: Server/Services/spotifyServices.js [hunk 3]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
...
+    .catch(error =>{
+        console.error(error)
+    })
+
+}
+
+/**
+ * Checks if the logged user is listening to a specific music searched by the music name
+ * @function isListeningTo
+ * @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:131]: missing async field

File: Server/Services/spotifyServices.js [hunk 4]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
...
+    })
+    .catch(error =>{
+        console.error(error)
+    })
+}
+
+/**
+ * If ther is a music playing, pause it
+ * @function pauseMusic
+ * @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:166]: missing async field

File: Server/Services/spotifyServices.js [hunk 5]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
...
+    .catch(error =>{
+        console.error(error)
+    })
+}
+
+/**
+* Set the shuffle mode to on or off, corresponding to want shuffle
+* The logged user should listen to music, or it can lead to undefined behavior
+* @function setShuffle
+* @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:192]: missing async field

File: Server/Services/spotifyServices.js [hunk 6]
```diff
@@ -61,6 +61,203 @@ var request = require('request');
...
+        accessToken : SpotifyTokens.accessToken,
+        refreshToken : SpotifyTokens.refreshToken,
+    }
+    firebaseFunctions.setDataInDb(`USERS/${SpotifyTokens.uid}/SpotifyService`, tokens)
+}
+
+/**
+ * Create a new playlist for the logged user
+ * @function createPlaylist
+ * @param {*} uid the uid of the spotify user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:230]: missing async field

File: Server/Services/spotifyServices.js [hunk 7]
```diff
@@ -71,7 +268,7 @@ module.exports = {
      * @param {*} res the request's result
      * @param {*} serverData client credentials of the developper account
```
Comment (osvegn) [inline Server/Services/spotifyServices.js]: missing spotifyToken parameters documentation

File: Server/Services/spotifyServices.js [hunk 8]
```diff
@@ -127,113 +326,43 @@ module.exports = {
...
-
     /**
-     * Checks if the logged user follows some artists
-     * @function isfollowing
-     * @param {*} req the request
-     * @param {*} res the request's result
-     * @param {*} artistUid the spotify uid of the searched artists
+     * @function SpotifyLoop
+     * @param {*} uid uid of the user
+     * @param {*} func function chosen by the user
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:327]: missing async field

File: Server/Services/spotifyServices.js [hunk 9]
```diff
@@ -127,113 +326,43 @@ module.exports = {
...
+                console.log(result)
+                resolve(result)
+            } else if (func === "shuffle") {
+                const result = setShuffle(uid)
+                console.log(result)
+                resolve(result)
+            } else if (func === "pause") {
+                const result = pauseMusic(uid)
+                console.log(result)
+                resolve(result)
```
Comment (osvegn) [inline Server/Services/spotifyServices.js:357]: There isn't better way to do that ?

File: Server/app.js [hunk 10]
```diff
@@ -153,6 +153,22 @@ class TwitchToken {
     }
 }
 
+/**
+ * @class class qui contient l'access token & le refresh token, ainsi que l'uid de l'utilisateur pour l'api spotify.
```
Comment (osvegn) [inline Server/app.js]: Documentation must be in english

**Review by osvegn — CHANGES_REQUESTED** [review]
Some changes to do

**Review by JohanCDev — APPROVED** [review]
Lgtm

**Review by osvegn — APPROVED** [review]

**Pull request merged**